src/sequencer.c

```diff
@@ -0,0 +1,193 @@
+#include "sequencer.h"
+
+typedef struct  {
+    struct event e; // parsed event -- we clear out sequence and time here obv
+    uint32_t tag;
+    uint32_t tick; // 0 means not used 
+    uint16_t divider; // 0 means not used 
+} sequence_entry;
+
+// linked list of sequence_entries
+typedef struct sequence_entry_ll_t{
+    sequence_entry *sequence;
```

Comment:
I think if you make this an actual sequence_entry rather than a (sequence_entry *), you'd only have to malloc once for each sequence_entry_ll, and you could remove a level of indirection, i.e. you'd be accessing (*entry_ll_ptr)->sequence.tag etc.